src/HealthChecks.Nats/DependencyInjection/NatsHealthCheckBuilderExtensions.cs

```diff
@@ -0,0 +1,50 @@
+using HealthChecks.Nats;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods to configure <see cref="NatsHealthCheck"/>.
+    /// </summary>
+    public static class NatsHealthCheckBuilderExtensions
+    {
+        internal const string NAME = "nats";
+
+        /// <summary>
+        /// Add a health check for Nats.
+        /// </summary>
+        /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+        /// <param name="setup">The action to configure the Nats setup.</param>
+        /// <param name="name">
+        /// The health check name.
+        /// Optional.
+        /// If <see langword="null"/> the type name 'nats' will be used for the name.
+        /// </param>
+        /// <param name="failureStatus">
+        /// The <see cref="HealthStatus"/> that should be reported when the health check fails.
+        /// Optional.
+        /// If <see langword="null"/> then the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+        /// </param>
+        /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+        /// <param name="timeout">An optional System.TimeSpan representing the timeout of the check.</param>
+        /// <returns>The <see cref="IHealthChecksBuilder"/>.</returns>
+        public static IHealthChecksBuilder AddNats(
+            this IHealthChecksBuilder builder,
+            Action<NatsOptions> setup,
+            string? name = default,
+            HealthStatus? failureStatus = default,
+            IEnumerable<string>? tags = default,
+            TimeSpan? timeout = default)
+        {
+            var options = new NatsOptions();
+            setup?.Invoke(options);
```

Comment:
```suggestion
            setup.Invoke(options);
```

I mean `setup` is required to check something useful except demo address "nats://demo.nats.io:4222",

Reply:
I'm OK to keep as is since parameter is not null. I see that other checks do the same.

Reply:
Commit: 0ab4b8af223791d4d882a83a2d82eaa9f0ab8cc9 [0ab4b8a]
Date: Tuesday, March 1, 2022 13:32:27